### PR TITLE
adding dependencies support for ubuntu 22

### PR DESCRIPTION
**Pull request by habibayassin**
Signed-off-by: habibayassin <habibayassin@aucegypt.edu>

### CODE REVIEW

File: etc/DependencyInstaller.sh [hunk 1]
```diff
@@ -5,26 +5,46 @@ set -euo pipefail
...
+        cmakeVersionSmall=${cmakeVersionBig}.0
+        swigVersion=4.0.1
+        swigChecksum="ef6a6d1dec755d867e7f5e860dc961f7"
+        boostVersionBig=1.76
+        boostVersionSmall=${boostVersionBig}.0
+        boostChecksum="e425bf1f1d8c36a3cd464884e74f007a"
+        eigenVersion=3.3
+        lemonVersion=1.3.1
+        lemonChecksum="e89f887559113b68657eca67cf3329b5"
+        spdlogVersion=1.8.1
```
Comment (vvbandeira) [inline etc/DependencyInstaller.sh:35]: why the slightly different versions? If the newest versions work for both versions of ubuntu we should keep them. @maliberty any reservations on changing the version of boost, eigen and spdlog?

Comment (habibayassin) [reply etc/DependencyInstaller.sh:35]: I will check the new versions compatibility with ubuntu 20.04

Comment (vvbandeira) [reply etc/DependencyInstaller.sh:35]: please also test with Centos7. you can use a docker image for quicker tests. thanks

Comment (maliberty) [reply etc/DependencyInstaller.sh:35]: I would prefer to have one version of dependencies on all platforms if possible.  My only concern is that spdlog 1.10 doesn't work for us.  If 1.9.2 compiles cleanly then I'm ok with it.

**Review by vvbandeira — APPROVED** [review]

**Pull request merged**